src/main/java/io/neonbee/health/HealthCheckRegistry.java

```diff
@@ -217,31 +234,40 @@ private List<Future<JsonArray>> sendDataRequests(JsonArray qualifiedNames, DataC
                 })).collect(toList());
     }
 
-    private JsonObject consolidateResults(List<JsonObject> healthCheckResults, DataContext dataContext) {
+    private JsonObject consolidateResults(List<JsonObject> healthCheckResults, String check, DataContext dataContext) {
         AtomicReference<String> aggregatedStatus = new AtomicReference<>(UP);
         Map<String, JsonObject> consolidatedChecks = new HashMap<>();
+        Predicate<String> isIncluded = c -> c.equals(check) || (c.startsWith("node") && c.endsWith(check));
+
+        healthCheckResults.stream()
+                .filter(checkResult -> Strings.isNullOrEmpty(check) || isIncluded.test(checkResult.getString(ID_KEY)))
+                .filter(checkResult -> checkResult.getString(ID_KEY) != null
+                        && checkResult.getString(STATUS_KEY) != null)
+                .forEach(checkResult -> {
+                    String checkId = checkResult.getString(ID_KEY);
+                    String status = checkResult.getString(STATUS_KEY);
+
+                    if (consolidatedChecks.containsKey(checkId)) {
+                        if (!status.equals(consolidatedChecks.get(checkId).getString(STATUS_KEY))) {
+                            LOGGER.correlateWith(dataContext).warn("Detected inconsistent status of health check {}",
+                                    checkId);
+                            // some nodes report a different status of the check than other nodes. This might be due to
+                            // some inconsistent uptime of the service, but more likely that some requirements to
+                            // perform a successful health check are not met by some nodes. Example: the health check
+                            // requests an external service which requires credentials, but the credentials are not
+                            // available on the node due to misconfiguration. In this scenario, keep the existing entry.
+                        }
+                    } else {
+                        consolidatedChecks.put(checkId, checkResult);
+                        if (DOWN.equals(status)) {
+                            aggregatedStatus.set(DOWN);
+                        }
+                    }
+                });
```

Comment:
I see no advantage in using a predicate here. I find it rather an unusual solution. In my opinion, it would be clearer to change the predicate to a method. I would also change the disjunction to two filter calls.

```
suggestion

        healthCheckResults.stream()
                .filter(checkResult -> Strings.isNullOrEmpty(check))
                .filter(checkResult -> isIncluded(check, checkResult))
                .filter(checkResult -> checkResult.getString(ID_KEY) != null
                        && checkResult.getString(STATUS_KEY) != null)
                .forEach(checkResult -> {
                    String checkId = checkResult.getString(ID_KEY);
                    String status = checkResult.getString(STATUS_KEY);

                    if (consolidatedChecks.containsKey(checkId)) {
                        if (!status.equals(consolidatedChecks.get(checkId).getString(STATUS_KEY))) {
                            LOGGER.correlateWith(dataContext).warn("Detected inconsistent status of health check {}",
                                    checkId);
                            // some nodes report a different status of the check than other nodes. This might be due to
                            // some inconsistent uptime of the service, but more likely that some requirements to
                            // perform a successful health check are not met by some nodes. Example: the health check
                            // requests an external service which requires credentials, but the credentials are not
                            // available on the node due to misconfiguration. In this scenario, keep the existing entry.
                        }
                    } else {
                        consolidatedChecks.put(checkId, checkResult);
                        if (DOWN.equals(status)) {
                            aggregatedStatus.set(DOWN);
                        }
                    }
                });

        if (LOGGER.isTraceEnabled()) {
            LOGGER.correlateWith(dataContext).trace("Consolidated health checks: " + consolidatedChecks);
        }

        return new JsonObject().put(CHECKS_KEY, new JsonArray(new ArrayList<>(consolidatedChecks.values())))
                .put(STATUS_KEY, aggregatedStatus.get()).put(OUTCOME_KEY, aggregatedStatus.get());
    }

    private boolean isIncluded(String check, JsonObject checkResult) {
        String c = checkResult.getString(ID_KEY);
        return c.equals(check) || (c.startsWith("node") && c.endsWith(check));
    }
```

Reply:
extracted it into a private method. changing the filter as suggested is not equivalent